Source/DedicatedServer/Private/ServerConsoleLinux.cpp

```diff
@@ -1,23 +1,246 @@
 // Copyright 2004-2016 YaS-Online, Inc. All Rights Reserved.
 
 #include "DedicatedServerPrivatePCH.h"
+#if PLATFORM_LINUX
+	#include <iostream>
+	#include <sstream>
+	#include <string>
+#endif
 
 void DumpConsoleHelp();
 
 #if WITH_SERVER_CODE && PLATFORM_LINUX
 	void FServerConsole::Tick()
 	{
+		int hInputEvents = getch();
+		if ( hInputEvents == ERR )
+		{
+			// Handle Error
+			return;
+		}
+		if ( hInputEvents == '\n' )
+		{
+			auto& sInput = m_sInput;
```

Comment:
```suggestion
			auto sInput = m_sInput;
```
Not sure if we need a reference here